### ISSUE TITLE
mysql introspection error

### ISSUE THREAD

**Issue by BennyAlex** [dbcodeio/public#124]
I am getting this error and I cant see the tables directly.
But writing Select statements works.

Error:
VT12001: unsupported: OVER CLAUSE with sharded keyspace

---
VSCode Version: 1.95.3
Extension Version: 1.4.12
OS: win32
Arch: x64
Database Version: PlanetScale MySQL Server - 8.0.34

**Comment by mikeburgh** [comment]
Thanks for reporting, looks like planet scale might have tweaked the MySQL query planner… I will get an instance spun up and see what's going on..

**Comment by mikeburgh** [comment]
I replicated the issue, and started to work around it, but there is another issue which causes an RPC error on PlanetScale, this query causes it.. and it's something to do with the nested JSON for the columns.

```sql
WITH tables AS (
    SELECT table_schema,table_name
    FROM information_schema.tables
    where table_schema = DATABASE()
    and table_type = 'BASE TABLE'
)
,columns as (
        select table_schema,table_name,column_name
	FROM information_schema.columns
	where table_schema = DATABASE()
)
select JSON_OBJECT(
       'tables', ( 
               select  COALESCE(JSON_ARRAYAGG(JSON_OBJECT(
               'type', 'table'
               ,'name', table_name
               ,'columns', (
                                Select COALESCE(JSON_ARRAYAGG(JSON_OBJECT(
                                        'type', 'column' ,'name',column_name
                                ) ),JSON_ARRAY())
                                from columns
                                where columns.table_name = tables.table_name
                        )
                )),JSON_ARRAY())
               from tables 
        ) 
) as t;
```


I raised the issue with Planetscale, to see what they come back with.. Guessing it wont be a quick fix, so I might need to look at re working the introspection approach totally for Planetscale.

**Comment by mikeburgh** [comment]
Planetscale have confirmed the issue (their support is excellent). They are saying it could be anywhere from a few weeks to months for the fix. So I will see how the workload goes on my end in the next week or two, and if I re work the introspection too solve this!

**Comment by mikeburgh** [comment]
I released version 1.5.4 which I re worked the introspection to remove the JSON, and hopefully work around the issues.

Can you give it a try and let me know how it goes ?

**Comment by mikeburgh** [comment]
I gave it a decent run through on a test DB in planetscale, seems to be working fine now, so going to close this. Re open if it's still happening for you.
